boilerplate/app/components/Icon.tsx

```diff
@@ -60,12 +60,17 @@ export function Icon(props: IconProps) {
     ...WrapperProps
   } = props
 
+  const isPressable = !!WrapperProps.onPress
   const Wrapper: ComponentType<TouchableOpacityProps> = WrapperProps?.onPress
     ? TouchableOpacity
     : View
 
   return (
-    <Wrapper {...WrapperProps} style={$containerStyleOverride}>
+    <Wrapper
+      {...WrapperProps}
+      accessibilityRole={isPressable ? "imagebutton" : "image"}
```

Comment:
I think there are a good amount of times where non-pressable icons or images are better off "invisible" to screen readers -- it depends whether they're decorative or conveying important meaning (e.g., ✔️ vs ❌ ). So I'm not sure this is something we want by default? If it is, at least `{...WrapperProps}` should probably come after these props so they can be overridden (that's probably generally true, not just for a11y)